cmd/thv-registry-api/app/serve.go

```
@@ -0,0 +1,164 @@
+package app
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	v1 "github.com/stacklok/toolhive/cmd/thv-registry-api/api/v1"
+	"github.com/stacklok/toolhive/cmd/thv-registry-api/internal/service"
+	thvk8scli "github.com/stacklok/toolhive/pkg/container/kubernetes"
+	"github.com/stacklok/toolhive/pkg/logger"
+)
+
+var serveCmd = &cobra.Command{
+	Use:   "serve",
+	Short: "Start the registry API server",
+	Long: `Start the registry API server to serve MCP registry data.
+The server reads registry data from ConfigMaps and provides REST endpoints for clients.`,
+	RunE: runServe,
+}
+
+const (
+	defaultGracefulTimeout = 30 * time.Second // Kubernetes-friendly shutdown time
+	serverRequestTimeout   = 10 * time.Second // Registry API should respond quickly
+	serverReadTimeout      = 10 * time.Second // Enough for headers and small requests
+	serverWriteTimeout     = 15 * time.Second // Must be > serverRequestTimeout to let middleware handle timeout
+	serverIdleTimeout      = 60 * time.Second // Keep connections alive for reuse
+)
+
+func init() {
+	serveCmd.Flags().String("address", ":8080", "Address to listen on")
+	serveCmd.Flags().String("configmap", "", "ConfigMap name containing registry data")
+
+	// Bind flags to viper
+	_ = viper.BindPFlag("address", serveCmd.Flags().Lookup("address"))
```

why errors are ignored here?